### PR TITLE
OCPCLOUD-1717: Add annotations to error output

### PR DESCRIPTION
**Pull request by racheljpg**
This is a PR for [OCPCLOUD-1717](https://issues.redhat.com/browse/OCPCLOUD-1717) to add annotations to error outputs in the AWS E2E tests to make debugging easier.

### CODE REVIEW

File: pkg/providers/aws.go [hunk 1]
```diff
@@ -35,18 +35,18 @@ var _ = Describe("MetadataServiceOptions", framework.LabelCloudProviderSpecific,
 	BeforeEach(func() {
 		var err error
 		client, err = framework.LoadClient()
-		Expect(err).ToNot(HaveOccurred())
+		Expect(err).ToNot(HaveOccurred(), "Failed to load client")
```
Comment (racheljpg) [inline pkg/providers/aws.go:38]: Should annotations like these be made more clear on which component they're failing to load, or is leaving them like this fine?

Comment (elmiko) [reply pkg/providers/aws.go:38]: as long as it is clear where the error originates from, and is not ambiguous with other error messages, then it's fine for me.

**Review by elmiko — COMMENTED** [review]
thanks @racheljpg !

/lgtm

**Pull request merged**